### PR TITLE
Add dependencies, add cli, fix bugs

### PR DESCRIPTION
**Pull request by rothoma2**
After testing it with some Researchers that are using this project, found a couple bugs. Did a few fixes:

- Added some missing dependencies, to the .toml file.
- Added an entrypoint script, so the user can invoke the tool just calling secret-harvest (use this for inspiration in other tools).
- Fix another bug, where for certain Keywords we might return a very large number of page, Github API returns only the first 1000 results.

### CODE REVIEW

File: secret_harvest/github_manager.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def search_code_repositories(self,
             for page in random_pages:
 
                 url = (f"https://api.github.com/search/repositories"
-                       f"?q={keyword}&page={page}")
+                       f"?q={keyword}")
```
Comment (rothoma2) [inline secret_harvest/github_manager.py:54]: Fix bug where page number might be very high. Github only provides first 1000 results.

File: secret_harvest/main.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import os
-from github_manager import GitHubManager
-from file_manager import FileManager
-from utility import Utility
+from secret_harvest.github_manager import GitHubManager
+from secret_harvest.file_manager import FileManager
+from secret_harvest.utility import Utility
```
Comment (rothoma2) [inline secret_harvest/main.py:5]: To use the entrypoint, the package name needs to be fully qualified/

File: pyproject.toml [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
 
 [tool.poetry.dependencies]
 python = "^3.10"
 requests = "^2.31.0"
+git-python = "^1.0.3"
+rich = "^13.7.1"
+tabulate = "^0.9.0"
 
+[tool.poetry.scripts]
+secret-harvest = "secret_harvest.main:main"
```
[the remark this answers is not in the document]
Comment (Patterbear) [reply pyproject.toml:16]: Looks good 👍🏼I used the same technique for docubleach

File: secret_harvest/main.py (hunk 2 above)
Comment (Patterbear) [reply secret_harvest/main.py:5]: Good stuff

**Pull request merged**